ELSpace/Commons/Presenters/Modal/Subviews/CustomIntensityVisualEffectView.swift

```diff
@@ -0,0 +1,15 @@
+import UIKit
+
+class CustomIntensityVisualEffectView: UIVisualEffectView {
+
+    init(effect: UIVisualEffect, intensity: CGFloat) {
+        super.init(effect: nil)
+        animator = UIViewPropertyAnimator(duration: 1, curve: .linear) { [unowned self] in self.effect = effect }
+        animator.fractionComplete = intensity
+    }
+
+    required init?(coder aDecoder: NSCoder) { return nil }
+
+    private var animator: UIViewPropertyAnimator!
+
+}
```

Comment:
What you think about deleting this class `CustomIntensityVisualEffectView`

and modify a little bit `ModalViewControllerBackgroundView`

```swift
import UIKit
import Anchorage

class ModalViewControllerBackgroundView: UIView {

    init() {
        super.init(frame: .zero)
        addSubview(blurView)
        blurView.edgeAnchors == edgeAnchors
        animator.startAnimation()
    }

    // MARK: - Subviews

    let blurView: UIVisualEffectView = UIVisualEffectView(effect: nil)

    private let blurEffect: UIBlurEffect = UIBlurEffect(style: .dark)

    private(set) lazy var animator: UIViewPropertyAnimator = {
        let animator = UIViewPropertyAnimator(duration: 1, curve: .linear) { [weak self] in
            self?.blurView.effect = self?.blurEffect
        }
        animator.fractionComplete = 0.18

        return animator
    }()

    // MARK: - Required initializers

    required init?(coder aDecoder: NSCoder) { return nil }

}
```

Reply:
I don't understand what value the change will bring - now we can use object of CustomIntensityVisualEffectView in a different place if needed.
After the change it would be merged together with the backgroundView so when sb wants to have slightly different backgroundView with another blur effect the code will be duplicated